invenio/templates/configurations/logstash.yaml

```diff
@@ -52,67 +52,133 @@ data:
         }
-        # add ua/patch only if set
-        if [@metadata][ua_patch] {
-            mutate {
-                add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}.%{[@metadata][ua_patch]}" }
-            }
-        } else {
+
+      useragent {
+          source => "[data][user_agent]"
+          target => "[@metadata]"
+      }
+      ##############################################################
+      # USER AGENT
+      ##############################################################
+      # add ua/device only if set
+      if [@metadata][ua_device] {
             mutate {
-                add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}" }
+                add_field => { "[data][ua/device]" => "%{[@metadata][ua_device]}" }
             }
         }
-        # add ua/os only if set
-        if [@metadata][ua_os] {
-            mutate {
-                add_field => { "[data][ua/os]" => "%{[@metadata][ua_os]}" }
-            }
+      # add ua/family only if set
+      if [@metadata][ua_name] {
+          mutate {
+              add_field => { "[data][ua/family]" => "%{[@metadata][ua_name]}" }
+          }
+      }
+      # add ua/patch only if set
+      if [@metadata][ua_patch] {
+          mutate {
+              add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}.%{[@metadata][ua_patch]}" }
+          }
+      } else {
+          mutate {
+              add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}" }
+          }
+      }
+      # add ua/os only if set
+      if [@metadata][ua_os] {
+          mutate {
+              add_field => { "[data][ua/os]" => "%{[@metadata][ua_os]}" }
+          }
+      }
+      # add a field with the category, based on the url
+      grok {
+         match => { "url" => ["^/(api/)?records?/%{DATA}/(media-)?files/"]}
+         add_field => {"[data][request_category]" => "files" }
+      }
+      # add a field with the category "legacy", based on the url
+      grok {
+         match => { "url" => ["^/communities/about/%{DATA}/", "^/collection/user-%{INT}", "^/communities/%{DATA}/about", "^/communities/%{DATA}/edit", "^/communities/%{DATA}/curate", "^/communities/%{DATA}/search", "^/collection/%{DATA}", "^/deposit/%{DATA} ", "^/record/%{DATA}", "^/record/%{DATA}/export/%{DATA}", "^/records/%{DATA}/export/%{DATA}", "^/record/%{DATA}/files/%{DATA}", "^/record/%{DATA}/preview/%{DATA}", "^/deposit$", "^/deposit/new$", "^/record/%{DATA}/thumb%{DATA}", "^/record/%{DATA}/formats%{DATA}", "^/dev$", "^/faq$", "^/features$", "^/whatsnew$", "^/about$", "^/contact$", "^/policies$", "^/privacy-policy$", "^/terms$", "^/donate$"]}
+         add_field => {"[data][request_category]" => "legacy" }
+      }
+      if "_grokparsefailure" in [tags] {
```

Comment:
This is tricky. Now, there are two grok (115 and 119). If any of them fail, we will hav the tag.

What about putting the grok for the legacy in line 128? (after the `if _grokparse`, and  before the `ui`).

Reply:
> This is tricky. Now, there are two grok (115 and 119). If any of them fail, we will hav the tag.
> 
> What about putting the grok for the legacy in line 128? (after the `if _grokparse`, and before the `ui`).

Is that an issue? Do we want the error tag for any reason?

Reply:
Well, we were using the tag to avoid overwritting the category if the previous tags had already matched.
 We could change the logic to see if the field is empty or not. 

The thing is that a grok that doesn't match (and we have here a couple) will, by default, add these tags (and those tags get written with the event at the end). If it is expected that it should not match, I think it is cleaner to avoid putting the tag.  That helps later on to debug in case there are other errors with grok